### PR TITLE
Paginate the CONFIG command's output

### PR DESCRIPTION
**Pull request by johnnovak**
# Description

As the title says. Displaying our built-in command's help with the `/?` or `-h` options does that already, so I've taught `CONFIG` the same trick.

It really bothered me that I had to make people type in things like `CONFIG -h glshader | MORE` in the getting started guide. Plus having to pipe things through `MORE` manually is just an annoyance for everybody.

I've also some made some further enhancements to the output (inserting some new lines here and there).

This fixes the PVS-Studio warnings too that I somehow introduced to `main`...

# Manual testing

Played around with the relevant `CONFIG` commands, and confirmed the output is now paginated.

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [ ] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Comment by FeralChild64** [comment]
Most likely the https://github.com/dosbox-staging/dosbox-staging/blob/jn/paginate-config-output/contrib/resources/translations/README.md file needs a small update. It claims:

> Do not exceed 79 characters on a line, or commands such as config -h fullscreen won't be able to display the help properly, and might wrap the text or display blank lines. The limit might be increased to 80 characters in the future, but the implementation is not there yet.

If I understand correctly, the implementation is now done (but I won't have time to test it today).

**Comment by FeralChild64** [comment]
Another general remark - I fully understand if you consider it out of scope of this PR, but it would be nice to also fix the wrap-around problem I have marked on the screenshot. ideally, the second line with _possible values_ should be indented:

Example

**Comment by johnnovak** [comment]
Thanks @FeralChild64, I'll take care of these two issues you mentioned in this PR. They seem simple enough.

**Comment by johnnovak** [comment]
@FeralChild64 80 char long lines are fine now, so I've updated the translation guide.

Only the `machine` setting has a long list of enum values, nothing else is affected. Frankly, I can't be bothered with that today. I've burned 20 minutes on it, but still not even halfway done, so I'm aborting it...

**Review by FeralChild64 — APPROVED** [review]

**Comment by FeralChild64** [comment]
That's perfectly fine - this is a minor issue, and not a regression.

File: src/misc/programs.cpp [hunk 1]
```diff
@@ -400,19 +400,14 @@ void CONFIG::HandleHelpCommand(const std::vector<std::string>& pvars_in)
 	}
 
 	Section_prop* psec = dynamic_cast<Section_prop*>(sec);
-	if (psec == nullptr) {
-		// Failed; maybe it's the autoexec section?
-		Section_line* pline = dynamic_cast<Section_line*>(sec);
-		if (pline == nullptr) {
-			E_Exit("Section dynamic cast failed.");
```
Comment (weirddan455) [inline src/misc/programs.cpp]: :+1: for getting rid of this.  No reason to kill the entire emulator because a command line program failed.

**Review by weirddan455 — APPROVED** [review]
This all looks good to me.

File: src/misc/programs.cpp (hunk 1 above)
Comment (johnnovak) [reply src/misc/programs.cpp]: Yeah, luckily many (but not all) of these hard exits are unreachable. I could not trigger this one, but yeah, using `E_Exit` should be extremely rare. I remove them on sight.

**Pull request merged**